source/common/filter/config_discovery_impl.h

```diff
@@ -490,7 +501,42 @@ class FilterConfigProviderManagerImpl : public FilterConfigProviderManagerImplBa
     return message;
   }
 
+  void storeFilterConfigs(const envoy::config::core::v3::TypedExtensionConfig& filter_config,
+                          const std::string& last_type_url, const std::string& last_version_info,
+                          const SystemTime& last_updated) override {
+    const std::string subscription_id = absl::StrCat(last_type_url, ".", filter_config.name());
+    // Remove the old config if existing.
+    auto it = filter_config_store_.find(subscription_id);
+    if (it != filter_config_store_.end()) {
+      filter_config_store_.erase(it);
+    }
+    // Add the filter config.
+    struct FilterConfigData filter_config_data;
+    filter_config_data.last_version_info_ = last_version_info;
+    filter_config_data.last_updated_ = last_updated;
+    filter_config_data.filter_config_ = filter_config;
+    filter_config_store_.insert({subscription_id, filter_config_data});
+  }
+
+  void removeFilterConfigs(const std::string& last_type_url,
+                           const std::string& filter_config_name) override {
+    const std::string subscription_id = absl::StrCat(last_type_url, ".", filter_config_name);
+    auto it = filter_config_store_.find(subscription_id);
+    if (it != filter_config_store_.end()) {
+      filter_config_store_.erase(it);
+    }
+  }
+
 private:
+  // Data structure to hold data related to filter configuration.
+  struct FilterConfigData {
+    std::string last_version_info_;
+    SystemTime last_updated_;
+    envoy::config::core::v3::TypedExtensionConfig filter_config_;
+  };
+  absl::flat_hash_map<std::string, struct FilterConfigData> filter_config_store_;
```

Comment:
I wonder how much memory this may end up using, especially if not used.
@jmarantz from your experience on the admin component, is memory consumption of saving raw protobufs something that we should take into consideration?
If so, will explicitly enabling/disabling the ECDS filter tracking the right way to go?

Reply:
This is a substantial concern for Wasm filters (which can be O(10MB) each). Although it's probably marginal on top of v8 overhead, but it does add O(10MB) at rutnime.

Reply:
Understood